### DISCUSSION TITLE
Suggestion: exit process with nonzero when `unknown flag` is thrown

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#2738] [Q&A] [answered]
Currently, `pocketbase` exists with `0` even when it bails out due to `unknown flag`. With https://pockethost.io, there is no point in running the instance again if it exits this way. Instead of checking `stderr`, I would prefer to just check the exit code. That way we can put the instance in maintenance mode (ie, won't run) until the user fixes the issue.

**Comment by ganigeorgiev** [comment, marked as answer]
It exists in my todo and I'll consider revisiting it with the planned logs generalization sometime in the near future. For now this is left aside because it complicates the regular Go code flow (_eg. returning `os.Exit(1)` will terminate immediately without performing the cleanup defer operations_).

**Reply by ganigeorgiev** [reply]
_Just for info in case someone stumble here: non-zero exit code support for commands returning an error (or panic) was added in the `develop` branch and will be available with the next v0.40.0 release (there are no ETAs)._
